src/main/java/com/minecolonies/util/BlockPosUtil.java

```diff
@@ -142,15 +142,29 @@ public static boolean isClose(@NotNull BlockPos coordinate, @NotNull EntityCitiz
      * @param block2 position two.
      * @return squared distance.
      */
-    public static int getDistanceSquared(@NotNull BlockPos block1, @NotNull BlockPos block2)
+    public static long getDistanceSquared(@NotNull BlockPos block1, @NotNull BlockPos block2)
     {
-        int i = block1.getX() - block2.getX();
-        int i1 = block1.getY() - block2.getY();
-        int i2 = block1.getZ() - block2.getZ();
+        long i = (long) block1.getX() - block2.getX();
+        long i1 = (long) block1.getY() - block2.getY();
+        long i2 = (long) block1.getZ() - block2.getZ();
         return i * i + i1 * i1 + i2 * i2;
     }
 
     /**
+     * 2D Squared distance between two BlockPos.
+     *
+     * @param block1 position one.
+     * @param block2 position two.
+     * @return 2D squared distance.
+     */
+    public static long getDistanceSquared2D(@NotNull BlockPos block1, @NotNull BlockPos block2)
+    {
+        long i = (long) block1.getX() - block2.getX();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'i' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)